include/numerics/type_tensor.h

```diff
@@ -320,6 +320,34 @@ class TypeTensor
   template <typename T2>
   const TypeTensor<T> & operator *= (const TypeTensor<T2> &);
 
+  /**
+   * Computes the matrix-matrix-matrix product, D = A * B * C, or
+   * D_ij = A_ik * B_kl * C_lj
+   * without creating a temporary. This makes sense as a static function
+   * since it does not require an object but needs to access the _coords
+   * member of the passed-in objects for efficient indexing operations.
+   */
+  static
+  TypeTensor<T>
+  mat_mult3(const TypeTensor<T> & A,
+            const TypeTensor<T> & B,
+            const TypeTensor<T> & C)
+  {
+    // Macro for readability
+#define IDX(i,j) i*LIBMESH_DIM + j
```

Comment:
Is this necessary?  Isn't this just what TypeTensor::operator() would be doing inline anyway?

Reply:
Hmm... for some reason I had it in my mind that operator(i,j) was doing bounds checking, but I see now that it's just a `libmesh_assert()`. So yeah, not necessary.